src/sudoku/singlarobin/react/component/Generate/index.jsx

```diff
@@ -0,0 +1,193 @@
+import { Fragment, useEffect, useState } from 'react';
+import Feature from '../Feature';
+import PlayArea from '../PlayArea';
+import classes from './styles.module.css';
+import { SUDOKU_API, DIFFICULTY } from '../../constants';
+import RadioButton from '../RadioButton';
+import Button from '../../../../../components/Button';
+import Loader from '../Loader';
+
+const Generate = () => {
+    const [game, setGame] = useState([]);
+    const [solution, setSolution] = useState([]);
+    const [undo, setUndo] = useState([]);
+    const [redo, setRedo] = useState([]);
+    const [count, setCount] = useState(0);
+    const [updatedIndex, setUpdatedIndex] = useState(-1);
+    const [started, setStarted] = useState(false);
+    const [loading, setLoading] = useState(false);
+    const [difficulty, setDifficulty] = useState('Easy');
+
+    const handleRadioButtonChange = (value) => setDifficulty(value);
+    const updatePlayArea = (game, index) => {
+        setGame(game);
+        setCount(count + 1);
+        setUpdatedIndex(index);
+
+        let sudoku = JSON.parse(localStorage.getItem('sudoku'));
+        setUndo(sudoku[2]['undo']);
+    };
+
+    useEffect(() => {
+        let sudoku = JSON.parse(localStorage.getItem('sudoku'));
+        if (sudoku !== null && sudoku.length !== 0) {
+            setSolution(sudoku[0]['solution']);
+            setGame(sudoku[1]['game']);
+            setUndo(sudoku[2]['undo']);
+            setRedo(sudoku[3]['redo']);
+            setStarted(true);
+        } else {
+            if (started) generateSudokuApiCall();
+        }
+    }, [started]);
+
+    const handleFormSubmit = () => setStarted(!started);
+
+    const handleUndo = () => {
+        if (undo.length === 0) {
+            alert('Not Possible');
+            return;
+        }
+        let operation = undo[undo.length - 1];
```

Comment:
if the value is not getting updated afterwards, then use const instead of let

Reply:
Done